### ISSUE TITLE
Feature: Allow ignoring undefined variables in file

### ISSUE THREAD

**Issue by gdavison** [terraform-linters/tflint#867]
If a variable is referenced in Terraform code, but not defined, `tflint` returns an error. Typically this is the correct response. However, there are a couple cases where we might want to skip this check:
* only a single file is being checked and the variable is defined in a different file
* the Terraform code being checked is a fragment, e.g. from documentation.

This would also make it behave similarly to missing resources or data sources, which do not cause an error to be returned.

Proposing the command-line flag `--ignore-undefined-vars`.

**Example**

From https://github.com/terraform-providers/terraform-provider-aws/blob/master/website/docs/r/cloudwatch_event_target.html.markdown#example-runcommand-usage

```terraform
resource "aws_cloudwatch_event_target" "stop_instances" {
  target_id = "StopInstance"
  arn       = "arn:aws:ssm:${var.aws_region}::document/AWS-RunShellScript"
  input     = "{\"commands\":[\"halt\"]}"
  rule      = aws_cloudwatch_event_rule.stop_instances.name
  role_arn  = aws_iam_role.ssm_lifecycle.arn

  run_command_targets {
    key    = "tag:Terminate"
    values = ["midnight"]
  }
}
```

Currently, this returns
```
Failed to check `aws_cloudwatch_event_target_invalid_arn` rule. An error occurred:

Error: Failed to eval an expression in /tmp/tmp.0Us0ePWN8H/main.tf:10; Reference to undeclared input variable: An input variable with the name "aws_region" has not been declared. This variable can be declared with a variable "aws_region" {} block.
```

With the flag set, this would not return any errors or issues.

**Comment by bendrucker** [comment]
Similar to #868, right? If the expression can't be evaluated, you could disable the rule.

**Comment by gdavison** [comment]
Yes, exactly
